file_handler.cpp

```diff
@@ -108,7 +131,27 @@ rep.content.append(buf, is.gcount());
 	else {
 		extension = "text/plain";
 	}
-	rep.headers[1].value = extension;
+
+	response->AddHeader("Content-Type", extension);
+
+	return RequestHandler::OK;
+}
+
+
+RequestHandler::Status file_handler::Init(const std::string& uri_prefix, const NginxConfig& config)
+{
+	uri_prefix_ = uri_prefix;
+
+	for(const auto& statement : config.statements_)
+	{
+		if(statement && statement->tokens_.size() >= 1)
+			if (statement->tokens_[0].compare(FILE_HANDLER_ROOT_TOKEN) == 0)
+				docroot = statement->tokens_[1];
+	}
+	if(!docroot.empty())
+		return RequestHandler::OK;
+	else
+		return RequestHandler::docroot_missing;
```

Comment:
Excellent job in having a check to see if the docroot is present or not. If not, we should definitely return an error.